backend/src/main/java/com/rikishi/backend/matching/systems/TreeBracket.java

```diff
@@ -39,16 +44,51 @@ public Set<Player> getCurrentPlayers() {
 
     @Override
     public void loadPlayers(Collection<Player> players) {
+        List<Player> playersArray = new ArrayList<>(players);
+        Collections.shuffle(playersArray);
         currentRound = OctetRoundName.FIRST_FIGHT;
         actualMatch = OctetRoundName.FIRST_FIGHT.getIndexStart();
         if (players.size() == 8) {
             for (int i = 0; i < players.size() * 2 - 1; i++) {
                 arrayTree.add(null);
             }
-            Iterator<Player> it = players.iterator();
+            Iterator<Player> it = playersArray.iterator();
             for (int i = 7; i < 15 && it.hasNext(); i++) {
                 arrayTree.set(i, it.next()); // Assign the next player
             }
-            } else throw new RuntimeException("implement only for 8 players");
+        } else if (5 < players.size() && players.size() < 8) {
+            for (int i = 0; i < 15; i++) {
+                arrayTree.add(null);
+            }
+            for (int i = 0,k=0; k < players.size(); i = ((i + 4) % 7), k++) {
+                arrayTree.set(7+i, playersArray.get(k));
+            }
+        } else throw new RuntimeException("implement only for 8 players");
```

Comment:
IMO nie wygląda to zbyt czytelnie, IMO lepiej
```java
if (...) {
    ...
} else {
    throw new ...
}
```

Reply:
IMO najczytelniej jest
if (size == 8) {
   ....
} else if (size == 6 || size == 7) {
   ...
} else{
   ...
}